nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -152,6 +153,14 @@
         .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
         .build();
 
+    public static final PropertyDescriptor CACHE_CONTROL = new PropertyDescriptor.Builder()
+            .name("Cache Control")
+            .displayName("Cache Control")
+            .description("Sets the Cache-Control HTTP header. Multiple directives are comma-separated.")
```

Comment:
Could you please add a similar explanation as in case of Content Type:
"Sets the Cache-Control HTTP header _indicating the caching directives of the associated object_." (or similar)

Reply:
Hi @turcsanyip Thank you for checking my PR. It's great to hear you ran the tests on it. I had carefully tested it on my local too.

In contrast to Content Type property, which can properly get MIME string from the flow file itself, the Cache Control is not relevant to flow file meta data, it is purely set to the processor itself.

Other similar properties such as Storage Class, Server Side Encryption ... are purely set to the processor and does not support expression language either.

I have updated the description as your suggestion. Thank you again.

Reply:
@kent-nguyen The generic approach is to add EL support with Variable Registry scope (`ExpressionLanguageScope.VARIABLE_REGISTRY`) at least and to support Flow File attributes ('ExpressionLanguageScope.FLOWFILE_ATTRIBUTES') when it is reasonable.
Storage Class property also has the option "Reference parameter..." (actually it is provided by the framework because it is a selectable property with a drop-down list, but in fact it is similar to setting Variable Registry scoped EL support from code).

As far as I see, you are not assigned the Contributor role yet. Could you please request for "Jira contributor access" on the mailing list dev@nifi.apache.org?
Then please assign the Jira ticket to yourself.